daemon/health.go

```diff
@@ -187,12 +187,19 @@ func handleProbeResult(d *Daemon, c *container.Container, result *types.Healthch
 func monitor(d *Daemon, c *container.Container, stop chan struct{}, probe probe) {
 	probeTimeout := timeoutWithDefault(c.Config.Healthcheck.Timeout, defaultProbeTimeout)
 	probeInterval := timeoutWithDefault(c.Config.Healthcheck.Interval, defaultProbeInterval)
+
+	intervalTimer := time.NewTimer(probeInterval)
+	defer intervalTimer.Stop()
+
 	for {
+		// make sure the timer is restarted on each loop
+		intervalTimer.Reset(probeInterval)
+
```

Comment:
Ideally that would be at the end of `for`, i.e.

```go
for {
        ........
        intervalTimer.Reset(probeInterval)
}
```

but I guess it's not very important here

Reply:
At the end of the loop we can't ensure this will run.